### PR TITLE
Add tests for improved candidate selection.

### PR DESCRIPTION
**Pull request by gafter**
Addresses remaining test items in #25235 for improved candidate selection.

/cc @cston

### CODE REVIEW

**Comment by gafter** [comment]
test windows_release_unit32_prtest please

**Comment by gafter** [comment]
@cston Could you please review these test-only additions for "improved candidate selection"?

**Comment by gafter** [comment]
test windows_release_vs-integration_prtest please

File: src/Compilers/CSharp/Test/Semantic/Semantics/BetterCandidates.cs [hunk 1]
```diff
@@ -81,6 +82,345 @@ class B {}
...
+class A {}
+class B {}
+";
+            CreateCompilationWithoutBetterCandidates(source, options: TestOptions.ReleaseExe).VerifyDiagnostics(
+                // (10,14): error CS0121: The call is ambiguous between the following methods or properties: 'Program.M(A)' and 'Program.M(B)'
+                //         this.M(null);
+                Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("Program.M(A)", "Program.M(B)").WithLocation(10, 14)
+                );
+            var compilation = CreateCompilationWithBetterCandidates(source, options: TestOptions.ReleaseExe).VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: "2");
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/BetterCandidates.cs]: Would it make sense to check the semantic model and checking candidate symbols? I'm assuming the static overload is not listed.

File: src/Compilers/CSharp/Test/Semantic/Semantics/BetterCandidates.cs [hunk 2]
```diff
@@ -81,6 +82,345 @@ class B {}
...
+@"class Program
+{
+    public static void Main()
+    {
+        Program p = new Program();
+        p.Main2();
+    }
+    void Main2()
+    {
+        this.M(null);
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/BetterCandidates.cs]: Consider testing `base.M(null);` as well.

Comment (gafter) [reply src/Compilers/CSharp/Test/Semantic/Semantics/BetterCandidates.cs]: Done.

**Comment by gafter** [comment]
test windows_release_unit64_prtest please

**Comment by jcouv** [comment]
test windows_release_unit32_prtest  please

**Comment by jcouv** [comment]
`Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense.VisualBasicCompletionCommandHandlerTests.TestProjections` failed again in windows_release_unit32_prtest ([details](https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_release_unit32_prtest/12633/))

@dotnet/roslyn-infrastructure FYI I'll re-run this leg

**Comment by jcouv** [comment]
test windows_release_unit32_prtest please

**Comment by gafter** [comment]
test windows_release_unit32_prtest please

**Pull request merged**